src/client/components/ui/Form/InputAutocomplétion/InputAutocomplétionMétier/InputAutocomplétionMétier.tsx

```diff
@@ -1,215 +1,128 @@
-			setMétierRecherchéInput(suggestionsApi[suggestionIndex].label);
-			setCodeRomesInput(suggestionsApi[suggestionIndex].romes);
-			setSuggestionsActive(false);
-			event.preventDefault();
-		}
-	};
-
-	const suggestions = useMemo(() => {
-		return <ul
-			className={styles.suggestionList}
-			role="listbox"
-			aria-labelledby={label}
-			id={CONTROL_ID}
-		>
-			{suggestionsApi.length === 0 &&
-          <li>Aucune proposition ne correspond à votre saisie. Vérifiez que votre saisie correspond bien à un métier.
-              Exemple : boulanger, ...
-          </li>
-			}
-			{suggestionsApi.map((suggestion, index) => (
-				<li
-					className={index === suggestionIndex ? styles.hover : ''}
-					key={index}
-					onMouseDown={(event) => {
-						handleClickOnSuggestion(event, suggestion);
-					}}
-					role="option"
-					aria-selected={suggestion.label === métierRecherchéInput}
-				>
-					{suggestion.label}
-				</li>
-			))}
-		</ul>;
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [suggestionsApi, label, suggestionIndex, métierRecherchéInput]);
+	}, [handleRechercherWithDebounce]);
 
 	return (
-		<div className={classNames(styles.wrapper, className)}>
+		<div>
 			{label && (
-				<label className={styles.label} htmlFor={inputId.current}>
+				<label className={styles.label} htmlFor={inputId}>
 					{label}
 				</label>
 			)}
-			<div ref={autocompleteRef}>
-				<div
-					role="combobox"
-					aria-expanded={suggestionsActive}
-					aria-controls={suggestionsActive ? CONTROL_ID : undefined}
-					aria-owns={suggestionsActive ? CONTROL_ID : undefined}
-					aria-haspopup="listbox"
-					onBlur={handleOnBlur}
-				>
-					<input
-						ref={inputRef}
-						className={classNames(styles.formControlInput, required && isTouched && !isValueValidSelected && styles.formControlInputError)}
-						type="text"
-						id={inputId.current}
-						name={name}
-						autoComplete="off"
-						aria-autocomplete="list"
-						aria-controls={suggestionsActive ? CONTROL_ID : undefined}
-						aria-activedescendant={inputId.current}
-						value={métierRecherchéInput}
-						onChange={
-							handleChange
-						}
-						onKeyDown={handleKeyDown}
-						required={required}
-						{...rest}
-					/>
-					<input type="hidden" value={codeRomesInput} name={'codeRomes'}/>
-				</div>
-				{suggestionsActive && suggestions}
-			</div>
-			{isTouched && inputRef.current?.validationMessage &&
-          <p id={errorId.current} className={styles.instructionMessageError}>{inputRef.current?.validationMessage}</p>}
+			<Combobox
+				ref={comboboxRef}
+				autoComplete="off"
+				id={inputId}
+				name={name}
+				valueName={'codeRomes'}
+				aria-label={label}
+				onChange={(event) => {
+					setFieldError(null);
+					handleRechercherWithDebounce(event.currentTarget.value);
+				}}
+				onInvalid={(event) => {
+					setFieldError(event.currentTarget.validationMessage);
+				}}
+				defaultValue={libellé}
+				requireValidOption
```

Comment:
c'est quoi la différence entre 'required' et 'requireValideOption' ?

Reply:
c'est quoi gerer le cas ou on veut quelque chose dans l'input mais pas forcement l'une des valeurs attendues ?

Reply:
C'est justement ça : 
- `requireValidOption` : le champ doit avoir une valeur parmi les options de la liste
- `required` : le champ doit avoir une valeur (pas forcément une de celles de la liste)

On s'est un peu creusé la tête avec DORO sur le nom de la props, _open to suggestions_

Reply:
pourquoi requireValidOption et pas requiredValidOption ?

Reply:
C'est ok en soit, tant qu'on le documente dans le stroybook 👍

Reply:
> pourquoi requireValidOption et pas requiredValidOption ?

C'est surtout une question de grammaire : 
_The field is **required**_
_The field **requires a valid option**_
Si on se place du point de vue du champ ça donne _"I require a valid option"_
🤷 

> C'est ok en soit, tant qu'on le documente dans le stroybook 👍

J'ai ajouté une ligne 👍